src/Models/SingleColumnModels/single_column_model.jl

```diff
@@ -4,151 +4,140 @@
 A single column model. Required fields are `domain`, `boundary_conditions`, and
 `parameters`.
 """
-Base.@kwdef struct SingleColumnModel{D, BC, P} <:
-                   Models.AbstractSingleColumnModel
+Base.@kwdef struct SingleColumnModel{D, B, T, M, F, BC, P} <:
+                   AbstractSingleColumnModel
     domain::D
+    base::B = AdvectiveForm()
+    thermodynamics::T = TotalEnergy()
+    moisture::M = Dry()
+    flux_corr::F = false
     boundary_conditions::BC
     parameters::P
 end
 
-function Models.variable_names(::SingleColumnModel)
-    base_vars = (:ρ, :uv, :w, :ρθ)
-    thermo_vars = (:ρθ,)
-    return (base = base_vars, thermodynamics = thermo_vars)
+function Models.components(model::SingleColumnModel)
+    (
+        base = model.base,
+        thermodynamics = model.thermodynamics,
+        moisture = model.moisture,
+    )
 end
 
 function Models.default_initial_conditions(model::SingleColumnModel)
-    space_c, space_f = make_function_space(model.domain)
-    local_geometry_c = CC.Fields.local_geometry_field(space_c)
-    local_geometry_f = CC.Fields.local_geometry_field(space_f)
-
-    # functions that make zeros for this model
-    zero_val = zero(CC.Spaces.undertype(space_c))
-    zero_scalar(lg) = zero_val
-    zero_12vector(lg) = CC.Geometry.UVVector(zero_val, zero_val)
-    zero_3vector(lg) = CC.Geometry.WVector(zero_val)
-
-    # base components
-    ρ = zero_scalar.(local_geometry_c)
-    uv = zero_12vector.(local_geometry_c)
-    w = zero_3vector.(local_geometry_f) # on faces
-
-    # thermodynamics components
-    ρθ = zero_scalar.(local_geometry_c)
+    # we need to provide default initial conditions for the model, because the ode solver
+    # requires inital conditions when getting instantiated, but we also want to support the `set!` function
+    # interface for initialization and re-initialization.
+    space_center, space_face = Domains.make_function_space(model.domain)
+    local_geometry_center = Fields.local_geometry_field(space_center)
+    local_geometry_face = Fields.local_geometry_field(space_face)
+
+    # initialize everything to zeros of the correct types on the correct spaces
+    FT = Spaces.undertype(space_center)
+    zero_inits = map(Models.components(model)) do component
```

Comment:
Is this function any different from the 2D / 3D models? I'm wondering if we can just unify them

Reply:
Possibly?  @bischtob

Reply:
#227